cpp/src/arrow/compute/kernels/cast_test.cc

```diff
@@ -604,6 +604,39 @@ TEST_F(TestCast, IntToFloatingPoint) {
 }
 #endif
 
+TEST_F(TestCast, DecimalToInt) {
+  CastOptions options;
+  options.allow_int_overflow = true;
+
+  std::vector<bool> is_valid = {true, true};
+  // simple cases decimal
```

Comment:
I would expect more test cases here, especially some examples where conversion fails because of overflow, but also examples including nulls.

Reply:
Added more tests: nulls, overflow, truncation